### PR TITLE
updated yamls for cloudcore incluster deployment

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
Signed-off-by: Kevin Wang <kevinwzf0126@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:
Update yamls to support running cloudcore as deployment inside K8s cluster

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update cloudcore in-cluster configs to v1.2.1
```

### CODE REVIEW

**Comment by fisherxu** [comment]
The service also need fixed, could we use hostNetwork by default for cloudcore deployment?

**Comment by kevin-wangzefeng** [comment]
Addressed review comments.

File: build/cloud/05-configmap.yaml [hunk 1]
```diff
@@ -7,33 +7,22 @@ metadata:
...
+    apiVersion: cloudcore.config.kubeedge.io/v1alpha1
+    kind: CloudCore
+    kubeAPIConfig:
+      kubeConfig: ""
+      master: ""
     modules:
-      enabled: [devicecontroller, edgecontroller, cloudhub]
+      cloudhub:
+        nodeLimit: 10
+        tlsCAFile: /etc/kubeedge/ca/rootCA.crt
```
Comment (fisherxu) [inline build/cloud/05-configmap.yaml]: It seems the `/etc/kubeedge/ca` also need mounted in container.

File: build/cloud/07-deployment.yaml [hunk 2]
```diff
@@ -79,5 +49,3 @@ spec:
       - name: certs
         secret:
           secretName: cloudcore
```
Comment (fisherxu) [inline build/cloud/07-deployment.yaml]:
```suggestion
      - name: certs
        secret:
          secretName: cloudcore
          items:
          - key: edge.crt
            path: certs
          - key: edge.key
            path: certs
          - key: rootCA.crt
            path: ca
```

File: build/cloud/05-configmap.yaml (hunk 1 above)
Comment (kevin-wangzefeng) [reply build/cloud/05-configmap.yaml]: rootCA.crt is mounted to /etc/kubeedge/certs/rootCA.crt
fixed

File: build/cloud/03-clusterrole.yaml [hunk 3]
```diff
@@ -6,47 +6,15 @@ metadata:
...
-  - delete
-  - get
-  - list
-  - patch
-  - update
-  - watch
-  
+- apiGroups: [""]
+  resources: ["nodes", "nodes/status", "configmaps", "pods", "pods/status", "secrets", "endpoints", "services"]
+  verbs: ["get", "list", "watch", "update"]
```
Comment (fisherxu) [inline build/cloud/03-clusterrole.yaml]: need create here for node

File: build/cloud/03-clusterrole.yaml [hunk 4]
```diff
@@ -6,47 +6,15 @@ metadata:
...
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["delete"]
+- apiGroups: ["devices.kubeedge.io"]
+  resources: ["devices", "devicemodels"]
```
Comment (fisherxu) [inline build/cloud/03-clusterrole.yaml]:
```suggestion
  resources: ["devices", "devicemodels", "devices/status", "devicemodels/status"]
```

File: build/cloud/03-clusterrole.yaml [hunk 5]
```diff
@@ -6,47 +6,15 @@ metadata:
...
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+- apiGroups: ["reliablesyncs.kubeedge.io"]
+  resources: ["objectsyncs", "clusterobjectsyncs"]
```
Comment (fisherxu) [inline build/cloud/03-clusterrole.yaml]:
```suggestion
  resources: ["objectsyncs", "clusterobjectsyncs", "objectsyncs/status", "clusterobjectsyncs/status"]
```

File: build/cloud/03-clusterrole.yaml (hunk 3 above)
Comment (kevin-wangzefeng) [reply build/cloud/03-clusterrole.yaml]:
```suggestion
  verbs: ["get", "list", "watch", "create", "update"]
```

File: build/cloud/07-deployment.yaml [hunk 6]
```diff
@@ -17,38 +17,10 @@ spec:
...
-              cluster: kubeedge
-              user: kubeedge
-          current-context: kubeedge
-          EOF
+      hostNetwork: true
       containers:
       - name: cloudcore
-        image: kubeedge/cloudcore:v1.1.0
+        image: kubeedge/cloudcore:v1.2.1
         imagePullPolicy: IfNotPresent
```
Comment (fisherxu) [inline build/cloud/07-deployment.yaml]:
```suggestion
        imagePullPolicy: Always
```

**Pull request merged**